src/transaction/create.go

```diff
@@ -144,6 +144,19 @@ func create(p Params, auxs coin.AddressUxOuts, headTime uint64, callCount int) (
 				toCoins[i] = to.Coins
 			}
 
+			// If sharefactor > 0 and sharehours < len(p.To), the sharehours cannot ensure each destination
+			// address has at least 1 coin hour. In this case, if we have enough hours, i.e the hours >= len(p.To),
+			// set the sharehours as len(p.To). If we don't have enough hours, i.e. hours < len(p.To),
+			// share all hours we have.
```

Comment:
This should consider if there is change, then the change output should be guaranteed at least one hour, if sharefactor < 1